### ISSUE TITLE
Bug in n_primitive_root_prime

### ISSUE THREAD

**Issue by vneiger** [flintlib/flint#1870]
The function `n_primitive_root_prime`, according to the doc, "Returns a primitive root for the multiplicative subgroup of Z/pZ where p is prime." There is no stated constraint on `p`, so one may assume this works for any `ulong` prime.

Its current implementation relies on `n_powmod_precomp`, which is designed for moduli `p` with $p < 2^{53}$. As a (non-surprising) result, `n_primitive_root_prime` is not correct when $p \ge 2^{53}$.

The funny thing is that this is not caught in the corresponding test file because it itself relies on the same `n_powmod_precomp` to test that the well-chosen powers of the computed primitive root are not 1. But since `n_powmod_precomp` does not correctly compute these powers, they are indeed never 1 (well, surely they could be in some rare cases), and the test passes where it should not.

The code below highlights this: its output shows why the test passes (`pow1` is not 1) although it should not (the actual correct power, `pow2`, is 1). Also the computed primitive root is 2, which is wrong (2 has order $(p-1) / 8$).

Output is
```
prime: 1139410705724735489
root: 2
pow1: 6813506898117712040
pow2: 1
```

Code is
```
#include <stdio.h>
#include <flint/ulong_extras.h>

int main()
{
    mp_limb_t p = 1139410705724735489;  // 60 bits prime, 1 + 2**52 * 11 * 23

    const mp_limb_t a = n_primitive_root_prime(p);

    double pinv1 = n_precompute_inverse(p);
    mp_limb_t pow1 = n_powmod_precomp(a, (p-1) / 2, p, pinv1);

    mp_limb_t pinv2 = n_preinvert_limb(p);
    mp_limb_t pow2 = n_powmod2_preinv(a, (p-1) / 2, p, pinv2);

    printf("prime: %ld\nroot: %ld\npow1: %ld\npow2: %ld\n", p, a, pow1, pow2);
}
```

**Comment by vneiger** [comment]
I'm going to fix this right now.
